packages/teleport/src/Welcome/NewCredentials/NewMfaDevice.tsx

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useMemo, useEffect } from 'react';
+import styled from 'styled-components';
+import { Text, ButtonPrimary, Flex, Box, Link, Image } from 'design';
+import { Danger } from 'design/Alert';
+import { ArrowBack } from 'design/Icon';
+import FieldInput from 'shared/components/FieldInput';
+import Validation, { Validator } from 'shared/components/Validation';
+import {
+  requiredField,
+  requiredToken,
+} from 'shared/components/Validation/rules';
+import createMfaOptions from 'shared/utils/createMfaOptions';
+import { SliderProps } from 'teleport/components/StepSlider';
+import { Props as CredentialsProps, LoginFlow } from './NewCredentials';
+import secKeyGraphic from './sec-key-with-bg.png';
+import { Auth2faType } from 'shared/services';
+
+export default function NewMfaDevice(props: Props) {
+  const {
+    resetToken,
+    submitAttempt,
+    clearSubmitAttempt,
+    auth2faType,
+    onSubmitWithWebauthn,
+    onSubmit,
+    password,
+    prev,
+    refCallback,
+  } = props;
+  const [otp, setOtp] = useState('');
+  const mfaOptions = useMemo(
+    () =>
+      createMfaOptions({
+        auth2faType: auth2faType,
+      }),
+    []
+  );
+  const [transitionPropertyName, setTransitionPropertyName] =
+    useState('height');
+  const [mfaType, setMfaType] = useState(mfaOptions[0]);
+  const [deviceName, setDeviceName] = useState(() =>
+    getDefaultDeviceName(mfaType.value)
+  );
+
+  useEffect(() => {
+    setTransitionPropertyName('');
+  }, []);
+
+  function onBtnClick(
+    e: React.MouseEvent<HTMLButtonElement>,
+    validator: Validator
+  ) {
+    e.preventDefault(); // prevent form submit default
```

Comment:
Is it required? From what I know it is needed for form elements (`<form onSubmit={}` />), but in this case it's just `onClick()`.